### PR TITLE
dragonball: metrics support for vsock device

### PR DESCRIPTION
**Pull request by lisongqian**
This PR adds metrics support for vsock device.

Fixes: #7248

Signed-off-by: Songqian Li <mail@lisongqian.cn>

### CODE REVIEW

File: src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs [hunk 1]
```diff
@@ -32,6 +32,57 @@ const VSOCK_CONFIG_SPACE_SIZE: usize = 8;
 const VSOCK_AVAIL_FEATURES: u64 =
     1u64 << uapi::VIRTIO_F_VERSION_1 | 1u64 << uapi::VIRTIO_F_IN_ORDER;
 
+/// Vsock-related metrics.
+#[derive(Default, Serialize)]
+pub struct VsockDeviceMetrics {
+    /// Number of times when handling events on a vsock device.
```
Comment (ZizhengBian) [inline src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs:38]: Can u add metrics to trace local init connection or peer init connection?

Comment (lisongqian) [reply src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs:38]: Of course, I've added them.

File: src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs [hunk 2]
```diff
@@ -32,6 +32,61 @@ const VSOCK_CONFIG_SPACE_SIZE: usize = 8;
...
+    /// How many times the killq has been resynced.
+    pub killq_resync: SharedIncMetric,
+    /// How many flush fails have been seen.
+    pub tx_flush_fails: SharedIncMetric,
+    /// How many write fails have been seen.
+    pub tx_write_fails: SharedIncMetric,
+    /// Number of times read() has failed.
+    pub rx_read_fails: SharedIncMetric,
+    /// Number of local init connections.
+    pub local_init: SharedIncMetric,
```
Comment (ZizhengBian) [inline src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs:85]: local_conn_init / peer_conn_init

**Review by ZizhengBian — APPROVED** [review]

File: src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs (hunk 2 above)
Comment (lisongqian) [reply src/dragonball/src/dbs_virtio_devices/src/vsock/device.rs]: Fixed.

**Pull request closed without merging**